docarray/array/mixins/find.py

```diff
@@ -87,14 +87,15 @@ def find(self: 'T', query: Dict, **kwargs) -> 'DocumentArray':
 
     def find(
         self: 'T',
-        query: Union['DocumentArray', 'Document', 'ArrayType', Dict],
+        query: Union['DocumentArray', 'Document', 'ArrayType', Dict, str, List[str]],
```

Comment:
I do not think this is desired. I think `Document` and `DocumentArray` are complete enough abstractions to handle what we want.